### PR TITLE
Allow custom location and fix issues

### PR DESCRIPTION
**Pull request by yupwei68**
1. allow custom location for the module:
fix :https://github.com/Azure/terraform-azurerm-compute/issues/133
https://github.com/Azure/terraform-azurerm-compute/issues/125

2. fix readme examples:
fix: https://github.com/Azure/terraform-azurerm-compute/issues/128

3. fix vm_os_simple count condition:
fix: https://github.com/Azure/terraform-azurerm-compute/issues/126

4. export availability zone of the vms:
fix:https://github.com/Azure/terraform-azurerm-compute/issues/91

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -42,3 +42,8 @@ output "availability_set_id" {
   description = "id of the availability set where the vms are provisioned."
   value       = azurerm_availability_set.vm.id
 }
+
+output "vm_zones" {
+  description = "a list of list of a single item of the Availability Zone which the Virtual Machine should be allocated in."
```
Comment (mybayern1974) [inline outputs.tf:47]: typo?

Comment (mybayern1974) [inline outputs.tf:47]:
```suggestion
  description = "a list of availability zones which the Virtual Machine should be allocated in."
```

Comment (mybayern1974) [reply outputs.tf:47]: Thank you @yupwei68 for making this! Overall LGTM with a typo issue. I'll merge it if you can address that.

**Comment by yupwei68** [comment]
@mybayern1974 Thanks for your comments. Corresponding changes have been pushed. Please continue reviewing.

File: outputs.tf [hunk 2]
```diff
@@ -42,3 +42,8 @@ output "availability_set_id" {
   description = "id of the availability set where the vms are provisioned."
   value       = azurerm_availability_set.vm.id
 }
+
+output "vm_zones" {
```
Comment (mybayern1974) [inline outputs.tf:46]: @yupwei68 , by re-going through issue #91 that this change serves, if there are more similar requests say prop A,B,C is expected to be outputted besides VM id, then the current design will be lack of extensibility. Is there any possibility to make a design that would be extensible?

Comment (yupwei68) [reply outputs.tf:46]: hi @mybayern1974 I would not suggest to change the existing outputs, cause it'll cause an unnecessary breaking change

**Pull request merged**